src/receivers/SocketModeReceiver.spec.ts

```diff
@@ -0,0 +1,288 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call, @typescript-eslint/naming-convention */
```

Comment:
Adding this test suite is a great start! 

In addition to the tests in this PR, we can add a test case to run `await receiver#stop()`. I know the test won't be a crucial test asset, but it slightly improves the code coverage. On the other hand, we cannot run `receiver#start()` as it blocks the test execution and the test results in a timeout error.

Also, mainly for your information, let me mention a few other things. As these are nice-to-haves in future development, you don't need to have anything further in this PR.

* Currently, we don't have any integration tests using both `App` and any of the built-in receivers. [App.spec.ts](https://github.com/slackapi/bolt-js/blob/main/src/App.spec.ts) uses only `FakeReceiver`, which does not do anything. We may want to have different ways to ensure the combination of App and a receiver works as we expect.

* (This may be more like an improvement on `@slack/socket-mode` side) In the future, we may want to have some tests verifying WebSocket communications (incl. auto-reconnection) and payload handling code. Currently, we don't have any tests for this, even on the `@slack/socket-mode` library side. Having some tests for the pattern will make future development / debugging reported issues.
  * For your reference, we have some tests and mock WS servers in Python/Java projects:
    * https://github.com/slackapi/bolt-python/blob/main/tests/adapter_tests/socket_mode/mock_socket_mode_server.py
    * https://github.com/slackapi/python-slack-sdk/blob/main/tests/slack_sdk/socket_mode/mock_socket_mode_server.py
    * https://github.com/slackapi/java-slack-sdk/blob/main/bolt-socket-mode/src/test/java/test_locally/SocketModeAppTest.java

I hope these are helpful to you (and anyone who read this comment!)